sim/priest/priest.go

```diff
@@ -18,11 +41,205 @@ func (priest *Priest) GetCharacter() *core.Character {
 func (priest *Priest) AddRaidBuffs(raidBuffs *proto.RaidBuffs) {
 	raidBuffs.DivineSpirit = proto.TristateEffect_TristateEffectRegular
 }
+
 func (priest *Priest) AddPartyBuffs(partyBuffs *proto.PartyBuffs) {
+
+}
+
+func (priest *Priest) Init(sim *core.Simulation) {
+	priest.mindflayCastTemplate = priest.newMindflayTemplate(sim)
+	priest.mindblastCastTemplate = priest.newMindBlastTemplate(sim)
+	priest.swpCastTemplate = priest.newSWPTemplate(sim)
+	priest.vtCastTemplate = priest.newVTTemplate(sim)
+	priest.swdCastTemplate = priest.newSWDTemplate(sim)
+}
+
+func (priest *Priest) Reset(newsim *core.Simulation) {
+	// Cleanup and pending dots and casts
+	priest.mindblastSpell = core.SimpleSpell{}
+	priest.swdSpell = core.SimpleSpell{}
+	priest.MindFlaySpell = core.SimpleSpell{}
+	priest.SWPSpell = core.SimpleSpell{}
+	priest.VTSpell = core.SimpleSpell{}
+
+	priest.Character.Reset(newsim)
 }
 
-func (priest *Priest) Reset(sim *core.Simulation) {}
+func (priest *Priest) Advance(sim *core.Simulation, elapsedTime time.Duration) {
+	// druid should never be outside the 5s window, use combat regen.
+	priest.Character.RegenManaCasting(sim, elapsedTime)
+	priest.Character.Advance(sim, elapsedTime)
+}
 
 func (priest *Priest) Act(sim *core.Simulation) time.Duration {
-	return core.NeverExpires // makes the bot wait forever and do nothing.
+	return core.NeverExpires // does nothing
+}
+
+func (priest *Priest) applyTalentsToShadowSpell(cast *core.Cast, effect *core.SpellHitEffect) {
+	if cast.ActionID.SpellID == SpellIDMF || cast.ActionID.SpellID == SpellIDMB {
```

Comment:
Would it make more sense to just put this logic with the code for these 2 spells? Don't have a strong opinion either way.